maps/mp/isnipe/extras/topplayer.gsc

```diff
@@ -0,0 +1,147 @@
+#include common_scripts\utility;
+#include maps\mp\_utility;
+#include maps\mp\gametypes\_hud_util;
+
+/**
+ * Initialize the topPlayer array on level.
+ * This is used to store data of the current top player.
+ *
+ * The following variables are stored:
+ * - level.topPlayer["playerName"]: this contains the name
+ *                                  of the current top player
+ * - level.topPlayer["playerGuid"]: this contains the guid
+ *                                  of the current top player
+ * - level.topPlayer["killDeathRatio"]: this contain the k/d ratio
+ *                                      of current top player
+ */
+init() {
+    if (!isDefined(level.topPlayer)) {
+        level.topPlayer = [];
+    }
+
+    _setSettingIfNotExists("playerName", "");
+    _setSettingIfNotExists("playerGuid", "");
+    _setSettingIfNotExists("killDeathRatio", 0);
+
+    self thread _updateRatios();
+}
+
+/**
+ * Set defaultValue in level.topPlayer[property] if it does not exist.
+ *
+ * @param property: the setting property to act upon
+ * @param defaultValue: the default value to set for the property if not set
+ */
+_setSettingIfNotExists(property, defaultValue) {
+    if (!isDefined(level.topPlayer[property])) {
+        level.topPlayer[property] = defaultValue;
+    }
+}
+
+/**
+ * Initialize the top player hud for a player.
+ */
+onPlayerConnect() {
+    self.topPlayerHud = [];
+}
+
+/**
+ * This method is called whenever a player spawns.
+ */
+onPlayerSpawned() {
+    self thread _drawHUD();
+    self thread _deleteHUD();
+}
+
+/**
+ * This method calculates the k/d ratio of each player and
+ * checks if it is higher than the current top player. If so
+ * the _registerNewTopPlayer method will be called for
+ * the current player.
+ */
+_updateRatios() {
+    for(;;){
+        foreach (player in level.players){
+
+            //Calculate our k/d ratio
+            if(player.deaths == 0){
+                player.ratio = player.kills;
+            } else {
+                player.ratio = player.kills / player.deaths;
+            }
+
+            //Check if ratio is higher than top player's ratio
+            if(player.ratio > level.topPlayer["killDeathRatio"]){
+                player _registerNewTopPlayer();
+            }
+
+            //This fixes the ratio not updating when the top player dies
+            if((player.ratio < level.topPlayer["killDeathRatio"]) && (player getguid() == level.topPlayer["playerGuid"])){
+                player _registerNewTopPlayer();
+            }
+
+        }
+        wait 0.2;
```

Comment:
If _updateRatios is updated to be on a player level and there is a notify event on the player which notifies at a kill, then you can wait till either a death or a kill of the player instead.

Reply:
There exists an `level player_spawned` notify, one for killing a player doesn't exist. Only player notify `killed_player` does.
You could of course make one yourself, by modifying `maps/mp/gametypes/_damage::PlayerKilled_internal` and putting it for example right under the `killed_player` notify.

Reply:
The `killed_player` notify sounds like a notify that the player has killed a player right? Or is it that the player is killed by another player?

Reply:
`killed_player` is called on the victim when he is killed by anything.

Reply:
I see, then that's not a viable option. But modifying like you said sounds like a good idea for this functionality